src/core/packages/application/browser-internal/src/scoped_history.test.ts

```diff
@@ -186,19 +186,26 @@ describe('ScopedHistory', () => {
   });
 
   describe('teardown behavior', () => {
-    it('throws exceptions after falling out of scope', () => {
+    it('throws exceptions after falling out of scope - only for action methods', () => {
       const gh = createMemoryHistory();
       gh.push('/app/wow');
       expect(gh.length).toBe(2);
       const h = new ScopedHistory(gh, '/app/wow');
       gh.push('/app/other');
-      expect(() => h.location).toThrowErrorMatchingInlineSnapshot(
+      expect(() => h.location).not.toThrow();
```

Comment:
nit: I think it would be clearer if we had separate it() tests for the read-only methods and other actions. So instead of a test structure like: 

'teardown behavior'
  -> 'throws exceptions after falling out of scope - only for action methods'

'after falling out of scope' 
  -> 'throws exceptions for non-read actions
  -> 'successfully returns values for read-only actions'

Reply:
Changed the tests according to this suggestion. [#a12f706](https://github.com/elastic/kibana/pull/220313/commits/a12f706de5c310c8c35bc1a0f08906e82063050e)